SignalServiceKit/src/Protos/Generated/WebSocketProto.swift

```diff
@@ -0,0 +1,328 @@
+//
+//  Copyright (c) 2018 Open Whisper Systems. All rights reserved.
+//
+
+import Foundation
+
+// WARNING: This code is generated. Only edit within the markers.
+
+public enum WebSocketProtoError: Error {
+    case invalidProtobuf(description: String)
+}
+
+// MARK: - WebSocketProtoWebSocketRequestMessage
+
+@objc public class WebSocketProtoWebSocketRequestMessage: NSObject {
+
+    // MARK: - WebSocketProtoWebSocketRequestMessageBuilder
+
+    @objc public class WebSocketProtoWebSocketRequestMessageBuilder: NSObject {
+
+        private var proto = WebSocketProtos_WebSocketRequestMessage()
+
+        @objc public override init() {}
+
+        @objc public func setVerb(_ valueParam: String) {
+            proto.verb = valueParam
+        }
+
+        @objc public func setPath(_ valueParam: String) {
+            proto.path = valueParam
+        }
+
+        @objc public func setBody(_ valueParam: Data) {
+            proto.body = valueParam
+        }
+
+        @objc public func addHeaders(_ valueParam: String) {
+            var items = proto.headers
+            items.append(valueParam)
+            proto.headers = items
+        }
+
+        @objc public func setRequestId(_ valueParam: UInt64) {
+            proto.requestId = valueParam
+        }
+
+        @objc public func build() throws -> WebSocketProtoWebSocketRequestMessage {
+            let wrapper = try WebSocketProtoWebSocketRequestMessage.parseProto(proto)
+            return wrapper
+        }
+    }
+
+    fileprivate let proto: WebSocketProtos_WebSocketRequestMessage
+
+    @objc public var verb: String? {
+        guard proto.hasVerb else {
+            return nil
+        }
+        return proto.verb
+    }
+    @objc public var hasVerb: Bool {
+        return proto.hasVerb
+    }
+
+    @objc public var path: String? {
+        guard proto.hasPath else {
+            return nil
+        }
+        return proto.path
+    }
+    @objc public var hasPath: Bool {
+        return proto.hasPath
+    }
+
+    @objc public var body: Data? {
+        guard proto.hasBody else {
+            return nil
+        }
+        return proto.body
+    }
+    @objc public var hasBody: Bool {
+        return proto.hasBody
+    }
+
+    @objc public var headers: [String] {
+        return proto.headers
+    }
+
+    @objc public var requestId: UInt64 {
```

Comment:
> I think this should probably be required.

Done.

> Given that it's apparently not required, why isn't this returning a UInt64?

Because:

> Property cannot be marked @objc because its type cannot be represented in Objective-C

Sorry, I was trying clumsily to explain this before.  I think our option here is to use `NSNumber?`, which would lose the numeric type precision.